TypeDBConsole.java

```diff
@@ -651,18 +656,43 @@ private void runQueryPrintAnswers(TypeDBTransaction tx, TypeQLQuery query) {
         if (query instanceof TypeQLDefine) {
             tx.query().define(query.asDefine()).get();
             printer.info("Concepts have been defined");
+            hasUncommittedChanges = true;
         } else if (query instanceof TypeQLUndefine) {
             tx.query().undefine(query.asUndefine()).get();
             printer.info("Concepts have been undefined");
+            hasUncommittedChanges = true;
         } else if (query instanceof TypeQLInsert) {
             Stream<ConceptMap> result = tx.query().insert(query.asInsert());
-            printCancellableResult(result, x -> printer.conceptMap(x, tx));
+            AtomicBoolean changed = new AtomicBoolean(false);
+            printCancellableResult(result, x -> {
+                changed.set(true);
+                printer.conceptMap(x, tx);
+            });
+            if (changed.get()) hasUncommittedChanges = true;
         } else if (query instanceof TypeQLDelete) {
-            tx.query().delete(query.asDelete()).get();
-            printer.info("Concepts have been deleted");
+            Stream<ConceptMap> result = tx.query().match(query.asDelete().match());
+            AtomicInteger answerCount = new AtomicInteger();
+            printCancellableResult(result, x -> {
+                answerCount.getAndIncrement();
+                printer.conceptMap(x, tx);
+            });
+            if (answerCount.get() > 0) {
+                tx.query().delete(query.asDelete()).get();
+                printer.info("Concepts have been deleted");
```

Comment:
I think you should just change this to `Deleted from <count> matched answers`

Reply:
I've changed it to: `Concepts have been deleted (answers matched: {count})`